build_modules/lib/src/module_library.dart

```diff
@@ -106,32 +143,64 @@ class ModuleLibrary {
     return ModuleLibrary._fromCompilationUnit(id, isEntryPoint, parsed);
   }
 
-  /// Parses the output of [toString] back into a [ModuleLibrary].
+  /// Parses the output of [serialize] back into a [ModuleLibrary].
   ///
   /// Importable libraries can be round tripped to a String. Non-importable
   /// libraries should not be printed or parsed.
-  static ModuleLibrary parse(AssetId id, String encoded) {
-    final lines = encoded.split('\n');
-    final isEntryPoint = lines.first == 'true';
-    final separator = lines.indexOf('');
-    final deps =
-        lines.sublist(1, separator).map((l) => AssetId.parse(l)).toSet();
-    final parts = lines
-        .sublist(separator + 1)
-        .where((l) => l.isNotEmpty)
-        .map((l) => AssetId.parse(l))
-        .toSet();
+  factory ModuleLibrary.deserialize(AssetId id, String encoded) {
+    var json = jsonDecode(encoded);
+
     return ModuleLibrary._(id,
-        isEntryPoint: isEntryPoint, deps: deps, parts: parts);
+        isEntryPoint: json['isEntrypoint'] as bool,
+        deps: _deserializeAssetIds(json['deps'] as Iterable),
+        parts: _deserializeAssetIds(json['parts'] as Iterable),
+        sdkDeps: Set.of((json['sdkDeps'] as Iterable).cast<String>()),
+        conditionalDeps:
+            (json['conditionalDeps'] as Iterable).map((conditions) {
+          return Map.of((conditions as Map<String, dynamic>)
+              .map((k, v) => MapEntry(k, AssetId.parse(v as String))));
+        }).toList());
   }
 
-  @override
-  String toString() => '${isEntryPoint ? 'true' : 'false'}\n'
-      '${deps.join('\n')}\n'
-      '\n'
-      '${parts.join('\n')}\n';
+  String serialize() => jsonEncode({
+        'isEntrypoint': isEntryPoint,
+        'deps': _deps.map((id) => id.toString()).toList(),
+        'parts': parts.map((id) => id.toString()).toList(),
+        'conditionalDeps': conditionalDeps
+            .map((conditions) =>
+                conditions.map((k, v) => MapEntry(k, v.toString())))
+            .toList(),
+        'sdkDeps': sdkDeps.toList(),
+      });
+
+  List<AssetId> depsForPlatform(Platform platform) {
+    return _deps.followedBy(conditionalDeps.map((conditions) {
+      var selectedImport = conditions[r'$default'];
+      assert(selectedImport != null);
+      for (var condition in conditions.keys) {
+        if (condition == r'$default') continue;
+        if (!condition.startsWith('dart.library.')) {
+          throw UnsupportedError(
+              '$condition not supported for config specific imports. Only the '
+              'dart.library.<name> constants are supported.');
+        }
+        var library = condition.substring('dart.library.'.length);
+        var coreLibSupport = platform.libraries[library];
```

Comment:
I'm not sure what's going on here with coreLibSupport.

Reply:
This is basically just checking if the current platform explicitly supports the core library that is being referenced in the conditional import. It might either omit the library entirely or explicitly have it marked as unsupported.